### PR TITLE
Callback in logout()

### PR DESCRIPTION
**Pull request by wmathurin**
This will allow application that use defer login to know when the logout has completed.
The js and the ios code needed to be changed but the android code was already expecting a callback.
There is a new basic template that does deferred login and demonstrates using the logout callback. See https://github.com/forcedotcom/SalesforceMobileSDK-Templates/pull/278

### CODE REVIEW

File: ios/SalesforceReact/SFOauthReactBridge.m [hunk 1]
```diff
@@ -53,6 +53,17 @@ @implementation SFOauthReactBridge
...
 {
     [SFSDKReactLogger d:[self class] format:@"logoutCurrentUser: arguments: %@", args];
+
+    __block id observerRef;
+    id observer = [[NSNotificationCenter defaultCenter]
+                   addObserverForName:kSFNotificationUserDidLogout
+                   object:nil
+                   queue:[NSOperationQueue mainQueue] usingBlock:^(NSNotification *note) {
+        callback(@[[NSNull null], @"OK"]);
+        [[NSNotificationCenter defaultCenter] removeObserver:observerRef];
```
Comment (wmathurin) [inline ios/SalesforceReact/SFOauthReactBridge.m:63]: We need to remove the observer once it fired.

Comment (bhariharan) [reply ios/SalesforceReact/SFOauthReactBridge.m:63]: Do we need this on `Android` too?

Comment (wmathurin) [reply ios/SalesforceReact/SFOauthReactBridge.m:63]: Android already does the right thing: https://github.com/forcedotcom/SalesforceMobileSDK-Android/blob/dev/libs/SalesforceReact/src/com/salesforce/androidsdk/reactnative/bridge/SalesforceOauthReactBridge.java#L83

File: src/react.force.oauth.js [hunk 2]
```diff
@@ -80,20 +74,9 @@ export const getAuthCredentials = (success, fail) => {
...
- * This method does not call back with a success or failure callback, as
- * (1) this method must not fail and (2) in the success case, the current user
- * will be logged out and asked to re-authenticate.  Note also that this method can only
- * be called once per page load.  Initiating logout will ultimately redirect away from
- * the given page (effectively resetting the logout flag), and calling this method again
- * while it's currently processing will result in app state issues.
+ * as well as any OAuth refresh token.  
  */
-export const logout = () => {
-    // only set callback for android, since iOS will not invoke callback.
```
Comment (wmathurin) [inline src/react.force.oauth.js:92]: The callback was not actually set for android despite the comment.

**Pull request merged**